### PR TITLE
Use Everest's state-registry API instead of manually creating them

### PR DESCRIPTION
**Pull request by psyGamer**
This reduces potential conflict with other mods which register states in the legacy way, and also allows dropping the CelesteTAS dependency to provide state information to the Info HUD.
I'm not sure which Everest version introduced that API, so just to be safe, depend on the latest stable.

### CODE REVIEW

File: src/CommunalHelper.csproj [hunk 1]
```diff
@@ -32,9 +32,6 @@
 		<Reference Include="$(CelestePrefix)\MMHOOK_Celeste.dll">
 			<Private>false</Private>
 		</Reference>
-		<Reference Include="$(CelestePrefix)\Celeste.dll">
```
Comment (bigkahuna443) [inline src/CommunalHelper.csproj]: Not sure why this was removed? Would prefer to have any build changes done in separate PRs

Comment (psyGamer) [reply src/CommunalHelper.csproj]: Because `Celeste.dll` is already included through the publicizer.
I dont't remember if it actually caused issue for me, but it's better to not include Celeste twice.

Comment (bigkahuna443) [reply src/CommunalHelper.csproj]: Gotcha. I'll undo it in this PR and include it in a separate commit.
